From 3073034e1d1adc6da4a771772f95505bd78105d9 Mon Sep 17 00:00:00 2001
From: Jaegeuk Kim <jaegeuk@kernel.org>
Date: Tue, 31 Jan 2017 16:24:35 -0800
Subject: [PATCH] f2fs + gapps support for 7.1

Should pico

Add in end of .repo/manifest.xml

  <remote name="opengapps" fetch="https://github.com/opengapps/"  />

  <project path="vendor/google/build" name="aosp_build" revision="master" remote="opengapps" />
  <project path="vendor/opengapps/sources/all" name="all" clone-depth="1" revision="master" remote="opengapps" />

  <!-- arm64 depends on arm -->
  <project path="vendor/opengapps/sources/arm" name="arm" clone-depth="1" revision="master" remote="opengapps" />
  <project path="vendor/opengapps/sources/arm64" name="arm64" clone-depth="1" revision="master" remote="opengapps" />

  <project path="vendor/opengapps/sources/x86" name="x86" clone-depth="1" revision="master" remote="opengapps" />
  <project path="vendor/opengapps/sources/x86_64" name="x86_64" clone-depth="1" revision="master" remote="opengapps" />

Signed-off-by: Jaegeuk Kim <jaegeuk@kernel.org>
---
 BoardConfig.mk    |  1 +
 aosp_angler.mk    |  2 +-
 device.mk         | 31 +++++++++++++++++++++++++++++++
 fstab.angler      |  3 ++-
 fstab.aosp_angler |  3 ++-
 5 files changed, 37 insertions(+), 3 deletions(-)

diff --git a/BoardConfig.mk b/BoardConfig.mk
index f003696..9025744 100644
--- a/BoardConfig.mk
+++ b/BoardConfig.mk
@@ -86,6 +86,7 @@ ifeq ($(HOST_OS),linux)
 endif
 
 TARGET_USERIMAGES_USE_EXT4 := true
+TARGET_USERIMAGES_USE_F2FS := true
 BOARD_BOOTIMAGE_PARTITION_SIZE := 33554432
 BOARD_RECOVERYIMAGE_PARTITION_SIZE := 33554432
 BOARD_SYSTEMIMAGE_PARTITION_SIZE := 3221225472
diff --git a/aosp_angler.mk b/aosp_angler.mk
index e59b338..0311538 100644
--- a/aosp_angler.mk
+++ b/aosp_angler.mk
@@ -29,7 +29,7 @@ PRODUCT_DEVICE := angler
 PRODUCT_BRAND := Android
 PRODUCT_MODEL := AOSP on angler
 PRODUCT_MANUFACTURER := Huawei
-PRODUCT_RESTRICT_VENDOR_FILES := true
+#PRODUCT_RESTRICT_VENDOR_FILES := true
 
 PRODUCT_COPY_FILES += device/huawei/angler/fstab.aosp_angler:root/fstab.angler
 
diff --git a/device.mk b/device.mk
index f9c479e..7d0fa6e 100644
--- a/device.mk
+++ b/device.mk
@@ -19,6 +19,33 @@
 #
 # Everything in this directory will become public
 
+GAPPS_VARIANT := stock
+
+GAPPS_FORCE_PACKAGE_OVERRIDES := true
+
+GAPPS_EXCLUDED_PACKAGES := \
+    PrebuiltExchange3Google \
+    PrebuiltGmail \
+    GoogleTTS \
+    PlusOne \
+    Hangouts \
+    Maps \
+    YouTube \
+    Books \
+    CloudPrint2 \
+    EditorsDocs \
+    Drive \
+    FitnessPrebuilt \
+    PrebuiltKeep \
+    Newsstand \
+    PrebuiltNewsWeather \
+    PlayGames \
+    EditorsSheets \
+    EditorsSlides \
+    talkback \
+    LatinImeGoogle \
+    TagGoogle
+
 # Enable support for chinook sensorhub
 TARGET_USES_CHINOOK_SENSORHUB := false
 
@@ -28,6 +55,8 @@ else
   LOCAL_KERNEL := $(TARGET_PREBUILT_KERNEL)
 endif
 
+BOARD_USERDATAIMAGE_FILE_SYSTEM_TYPE := f2fs
+
 PRODUCT_COPY_FILES := \
     $(LOCAL_KERNEL):kernel
 
@@ -490,3 +519,5 @@ endif
 # b/29995499
 $(call add-product-sanitizer-module-config,cameraserver,never)
 $(call add-product-sanitizer-module-config,mm-qcamera-daemon,never)
+
+$(call inherit-product, vendor/google/build/opengapps-packages.mk)
diff --git a/fstab.angler b/fstab.angler
index de0e0b1..8bb9127 100644
--- a/fstab.angler
+++ b/fstab.angler
@@ -5,7 +5,8 @@
 
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/system       /system         ext4    ro,barrier=1,inode_readahead_blks=8                             wait,verify=/dev/block/platform/soc.0/f9824900.sdhci/by-name/metadata
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/vendor       /vendor         ext4    ro,barrier=1,inode_readahead_blks=8                             wait,verify=/dev/block/platform/soc.0/f9824900.sdhci/by-name/metadata
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata     /data           ext4    noatime,nosuid,nodev,barrier=1,data=ordered,nomblk_io_submit,noauto_da_alloc,discard,errors=panic,inode_readahead_blks=8 wait,check,forcefdeorfbe=/dev/block/platform/soc.0/f9824900.sdhci/by-name/metadata
+#/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata     /data           ext4    noatime,nosuid,nodev,barrier=1,data=ordered,nomblk_io_submit,noauto_da_alloc,discard,errors=panic,inode_readahead_blks=8 wait,check,forcefdeorfbe=/dev/block/platform/soc.0/f9824900.sdhci/by-name/metadata
+/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata     /data           f2fs    noatime,nosuid,nodev,discard wait,check
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/cache        /cache          ext4    noatime,nosuid,nodev,barrier=1,data=ordered,nomblk_io_submit,noauto_da_alloc,errors=panic wait,check
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/persist      /persist        ext4    nosuid,nodev,barrier=1,data=ordered,nodelalloc,nomblk_io_submit,errors=panic wait,notrim
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/modem        /firmware       vfat    ro,shortname=lower,uid=1000,gid=1000,dmask=227,fmask=337,context=u:object_r:firmware_file:s0        wait
diff --git a/fstab.aosp_angler b/fstab.aosp_angler
index ab57d3a..d7683d6 100644
--- a/fstab.aosp_angler
+++ b/fstab.aosp_angler
@@ -5,7 +5,8 @@
 
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/system       /system         ext4    ro,barrier=1,inode_readahead_blks=8                             wait,verify=/dev/block/platform/soc.0/f9824900.sdhci/by-name/metadata
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/vendor       /vendor         ext4    ro,barrier=1,inode_readahead_blks=8                             wait
-/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata     /data           ext4    noatime,nosuid,nodev,barrier=1,data=ordered,nomblk_io_submit,noauto_da_alloc,discard,errors=panic,inode_readahead_blks=8 wait,check,forcefdeorfbe=/dev/block/platform/soc.0/f9824900.sdhci/by-name/metadata
+#/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata     /data           ext4    noatime,nosuid,nodev,barrier=1,data=ordered,nomblk_io_submit,noauto_da_alloc,discard,errors=panic,inode_readahead_blks=8 wait,check,forcefdeorfbe=/dev/block/platform/soc.0/f9824900.sdhci/by-name/metadata
+/dev/block/platform/soc.0/f9824900.sdhci/by-name/userdata     /data           f2fs    noatime,nosuid,nodev,discard wait,check
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/cache        /cache          ext4    noatime,nosuid,nodev,barrier=1,data=ordered,nomblk_io_submit,noauto_da_alloc,errors=panic wait,check
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/persist      /persist        ext4    nosuid,nodev,barrier=1,data=ordered,nodelalloc,nomblk_io_submit,errors=panic wait,notrim
 /dev/block/platform/soc.0/f9824900.sdhci/by-name/modem        /firmware       vfat    ro,shortname=lower,uid=1000,gid=1000,dmask=227,fmask=337,context=u:object_r:firmware_file:s0        wait
-- 
1.9.1

